apps/dot-cdn/src/app/dotcdn.store.ts

```diff
@@ -0,0 +1,105 @@
+import { Injectable } from '@angular/core';
+import { ComponentStore } from '@ngrx/component-store';
+import { Observable } from 'rxjs';
+import { ChartData, DotChartStats } from './app.interface';
+
+export interface DotCDNState {
+    chartData: ChartData;
+    statsData: DotChartStats[];
+    isChartLoading: string;
+    isPurgeLoading: string;
+    isPurgeZoneLoading: string;
+}
+
+export const enum LoadingState {
+    IDLE = 'IDLE',
+    LOADING = 'LOADING',
+    LOADED = 'LOADED'
+}
+
+export const enum Loader {
+    CHART = 'CHART',
+    PURGE_URLS = 'PURGE_URLS',
+    PURGE_PULL_ZONE = 'PURGE_PULL_ZONE'
+}
+
+@Injectable()
+export class DotCDNStore extends ComponentStore<DotCDNState> {
+    constructor() {
+        super({
+            chartData: {
+                labels: [],
+                datasets: []
+            },
+            statsData: [],
+            isChartLoading: LoadingState.IDLE,
+            isPurgeLoading: LoadingState.IDLE,
+            isPurgeZoneLoading: LoadingState.IDLE
+        });
+    }
+
+    readonly chartData$: Observable<ChartData> = this.select((state) => state.chartData);
+    readonly statsData$: Observable<DotChartStats[]> = this.select((state) => state.statsData);
+    readonly isChartLoading$: Observable<boolean> = this.select(
+        (state) => state.isChartLoading === LoadingState.LOADING
+    );
+
+    readonly isPurgeUrlsLoading$: Observable<boolean> = this.select(
+        (state) => state.isPurgeLoading === LoadingState.LOADING
+    );
+
+    readonly isPurgeZoneLoading$: Observable<boolean> = this.select(
+        (state) => state.isPurgeZoneLoading === LoadingState.LOADING
+    );
```

Comment:
You should group all this in a `vm$` property and then just have ONE property in the component to print all using

```html
<ng-container *ngIf="vm$ | async as VM">
    {{vm. chartData}} <!-- and so on. -->
</ng-container>